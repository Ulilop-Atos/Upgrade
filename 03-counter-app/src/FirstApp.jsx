import { PropTypes } from "prop-types";


const FirstApp = ({ hiFrom, subtitle }) => {
    
    let hi = `Hello World`;
    hi += hiFrom? `, I'm ${hiFrom}`: '';
    
    // <> is a shortcut for fragments 
    return <>
        <h1>{hi}</h1>
        {/* <p>{ JSON.stringify({ name:'Fer'}, null, 3) }</p> */}
        <p>{subtitle}</p>
    </>;
}

FirstApp.propTypes = {
    hiFrom: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

FirstApp.defaultProps = {
    subtitle: 'This is a Subtitle',
}

export default FirstApp;

