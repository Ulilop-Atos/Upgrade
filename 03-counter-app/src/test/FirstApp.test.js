import { render } from "@testing-library/react";
import { shallow } from "enzyme";
import FirstApp from "../FirstApp";

describe('Test <FirstApp ></FirstApp >', ()=>{

    // Test using vanilla Jest
    test(`Should show the message "Hello World, I'm, Goku"`, () => {
        const content = `Hello World, I'm Goku`;
        // This is a wrapper
        const {getByText} = render(<FirstApp hiFrom={'Goku'}/>);
        expect( getByText(content) ).toBeInTheDocument();

    });

    // Test using Enzime
    test(`Should show <FirstApp /> correctly`, () => {
        
        const title = `Hello World, I'm Goku`;
        const subtitle = `I'm a strong warrior!`;

        const wrapper = shallow(<FirstApp 
            hiFrom={`Goku`}
            subtitle={subtitle}
            />);
            
        expect( wrapper ).toMatchSnapshot();

        expect(wrapper.find('h1').text()).toBe(title);
        expect(wrapper.find('p').text()).toBe(subtitle);
    });
});