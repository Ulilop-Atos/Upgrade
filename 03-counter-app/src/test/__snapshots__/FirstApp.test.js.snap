// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FirstApp ></FirstApp > Should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hello World, I'm Goku
  </h1>
  <p>
    I'm a strong warrior!
  </p>
</Fragment>
`;
